### PR TITLE
LPS-108987

### PR DESCRIPTION
**Pull request by wincent**
This replaces:

- https://github.com/wincent/liferay-portal/pull/156 (and its follow-up, https://github.com/wincent/liferay-portal/pull/159)
- https://github.com/wincent/liferay-portal/pull/157 (and its follow-up, https://github.com/wincent/liferay-portal/pull/160)

I basically rebased those PRs, resolved conflicts, tried to sort out conflicting/duplicate sections, and split them up (very roughtly) according to the related LPS tickets (data-engine JSP, data-engine React, journal-web changes). I also added a commit removing the now-dead code of the `useSidebarContent` hook.

I'll use this as a new basis for building and testing.

### CODE REVIEW

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
Rebased on top of https://github.com/brianchandotcom/liferay-portal/pull/85804 (which itself has conflicts, but will certainly go in before this one does). Will need additional fixes to actually work though.

File: modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/sidebar/MultiPanelSidebar.es.js [hunk 1]
```diff
@@ -199,32 +199,28 @@ export default function MultiPanelSidebar({panels, sidebarPanels}) {
 									pluginEntryPoint
 								).then(...swallow);
 
-							return (
-								<>
```
Comment (jbalsas) [inline modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/sidebar/MultiPanelSidebar.es.js]: @diegonvs, looks like this was the `key` issue!! 😱

**Comment by ealonso** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/sidebar/MultiPanelSidebar.es.js (hunk 1 above)
Comment (diegonvs) [reply modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/sidebar/MultiPanelSidebar.es.js]: W e i r d

Comment (wincent) [reply modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/sidebar/MultiPanelSidebar.es.js]: Not weird at all. The error message (which you can see in [the commit message](https://github.com/wincent/liferay-portal/pull/162/commits/992ab3b310576e18b8e069fb34ae858becee8551)) is very explicit:("Warning: Each child in a list should have a unique "key" prop") and it even told you which component was responsible (`<MultiPanelSideBar />`).

This is an example of where `<MultiPanelSidebar />` diverges from the original `<Sidebar />` component in the page editor (the page editor doesn't have the key warning).

**Comment by wincent** [comment]
Going to update this with the new commits from: https://github.com/wincent/liferay-portal/pull/167

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:stable

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
Will keep an eye on the resend of https://github.com/brianchandotcom/liferay-portal/pull/85852 because some of the commits here will head upstream in a probably different form. In the meantime, I am going to look into the app-builder layout breakage.

**Comment by wincent** [comment]
Just did a quick test in the Chrome inspector of the idea I suggested in https://github.com/wincent/liferay-portal/pull/162/commits/cfd5dd550f5deb951421ee5696bbc5d1cdb3e60f:

> - Mount sidebar via a React portal inside the top `position: sticky` nav. Put it at the bottom of the nav inside a `position: relative` container with zero height.
> - Take sidebar out of flow using `position: absolute` and `top: 0`.
> - Make nav `overflow: visible` so that we can see the sidebar that is actually outside of it.
> - Set sidebar height with combination of `calc` and `vh`, and if that doesn't work, fallback to doing a JS-based calculation as a last resort.

Took off `position: fixed` from the sidebar, just to show where it is in the natural DOM order with static positioning:

fixed

Copied the markup into a 0-height wrapper (`position: relative`) at  the bottom of `nav`. Made contents `position: absolute` to take them out of flow and ensure `overflow: visible` so that we can see them:

moved

As you can see, even after removing all `top:` etc the sidebar renders in the right place. Height is currently coming due to height of content, but even if we can't make it work with some combination of `calc` and `vh` etc the sidebar is always a React component so we have the option of running JS to do a runtime calculation as a last resort.

(Buttons are on the left because I removed the previous `position` on them, but that would be easily fixed in the real version.)

Seems like too big a change to go for in this PR, but certainly something to look into when we do next steps.

**Comment by ealonso** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ealonso** [comment]
see: https://github.com/wincent/liferay-portal/pull/169

I did the following changes:
- I rebased from: https://github.com/brianchandotcom/liferay-portal/pull/85895, since it is already on master
- Group commits by LPS
- Split the first commits to make more clear which files are auto generated using buildTaglib
- Change the commit author to the real one, 😂, @jbalsas 
- Fix SF errors
- Update spacing on app builder
- Removes unnecessary files on Journal
- Implements multi panels on DL, it should be send to @adolfopa to further review, since it is not complete

Things to take into account:
- We should have the ability of setting the order of the panels

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
ci:test:stable

**Comment by wincent** [comment]
ci:test:forms

**Comment by wincent** [comment]
ci:test:app-builder

**Comment by eduardoallegrini** [comment]
Just started reviewing :)

**Comment by adolfopa** [comment]
I've sent the parts of this that touch DM to Brian. See https://github.com/brianchandotcom/liferay-portal/pull/86020.

**Comment by wincent** [comment]
Thanks @adolfopa!

**Comment by ealonso** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ealonso** [comment]
see: https://github.com/brianchandotcom/liferay-portal/pull/86032

**Comment by ealonso** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by eduardoallegrini** [comment]
I took a look at the CSS code responsible for the positioning and behavior of the multi-panel-sidebar. I think that at the moment we can't really work on it because the real component is still in development in Clay https://github.com/liferay/clay/issues/2903 But since we can anticipate the positioning problems that this components will probably have, I've been studying and preparing alternatives ways for it.

- Original (temporary broken) https://codepen.io/eduardoallegrini/pen/RwNqQPB
- Grid (not supported by ie) https://codepen.io/eduardoallegrini/pen/ExjPppv
- JS https://codepen.io/eduardoallegrini/pen/bGdaZVY

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:stable

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
So I rebased this and tried testing it but app-builder seems pretty broken right now, so I can't. This is a screenshot from master, not this PR:

Custom_Objects_-_Liferay_DXP

**Comment by wincent** [comment]
Closing this. Rebased and forwarded here: https://github.com/brunobasto/liferay-portal/pull/1153

**Pull request closed without merging**